### PR TITLE
Some cleanups

### PR DESCRIPTION
**Pull request by owenthereal**
- Use more asserts in tests
- Move Request#Do to its on file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.test
```
Comment (technoweenie) [inline .gitignore]: What is this for?

File: request_test.go [hunk 2]
```diff
@@ -27,14 +27,10 @@ func TestSuccessfulGet(t *testing.T) {
 	apierr := &TestError{}
 
 	req, err := client.NewRequest("user", apierr)
-	if err != nil {
-		t.Fatalf("request errored: %s", err)
-	}
+	assert.Equal(t, nil, err)
```
Comment (technoweenie) [inline request_test.go:30]: Have you verified this works?  Nil checking with [DeepEqual](https://github.com/bmizerany/assert/blob/master/assert.go#L30) doesn't work reliably in my experience.

EDIT: Totally meant DeepEqual, not DeepCopy :)

Comment (owenthereal) [inline request_test.go:30]: All tests passed. Or you're not expecting DeepEqual here?

Comment (technoweenie) [inline request_test.go:30]: assert.Equal uses DeepEqual.  It works fine with error objects, but not other types (which is what I had issues with): http://play.golang.org/p/Tw90V9ZwtA

So this is all legit, but I'd like to fork assert and add `assert.Nil` and `assert.NotNil`.

**Comment by technoweenie** [comment]
Overall :+1:

File: .gitignore (hunk 1 above)
Comment (owenthereal) [inline .gitignore]: My vim automatically compiles Go test when I type "w". I moved the gitignore to my global settings. Sorry for the confusion

File: request_test.go (hunk 2 above)
Comment (owenthereal) [inline request_test.go:30]: Good call on the fork

**Pull request merged**